samples/gen-js/index.js

```diff
@@ -200,23 +202,179 @@ class SwaggerTest {
+   * @returns {function} iter.forEach - takes in a function, applies it to each resource
+   */
+  getAuthorsIter(params, options) {
+    const it = (f, saveResults, cb) => new Promise((resolve, reject) => {
+      const rejecter = (err) => {
+        reject(err);
+        if (cb) {
+          cb(err);
+        }
+      };
+      const resolver = (data) => {
+        resolve(data);
+        if (cb) {
+          cb(null, data);
+        }
+      };
+
+
+      if (!options) {
+        options = {};
+      }
+
+      const timeout = options.timeout || this.timeout;
+      const span = options.span;
+
+      const headers = {};
+
+      const query = {};
+      if (typeof params.name !== "undefined") {
+        query["name"] = params.name;
+      }
+  
+      if (typeof params.startingAfter !== "undefined") {
+        query["startingAfter"] = params.startingAfter;
+      }
+  
+
+      if (span) {
+        opentracing.inject(span, opentracing.FORMAT_TEXT_MAP, headers);
+        span.logEvent("GET /v1/authors");
+        span.setTag("span.kind", "client");
+      }
+
+      const requestOptions = {
+        method: "GET",
+        uri: this.address + "/v1/authors",
+        json: true,
+        timeout,
+        headers,
+        qs: query,
+        useQuerystring: true,
+      };
+  
+
+      const retryPolicy = options.retryPolicy || this.retryPolicy || singleRetryPolicy;
+      const backoffs = retryPolicy.backoffs();
+  
+      let results = [];
+      async.whilst(
+        () => requestOptions.uri !== "",
+        cbW => {
+      const address = this.address;
+      let retries = 0;
+      (function requestOnce() {
+        request(requestOptions, (err, response, body) => {
+          if (retries < backoffs.length && retryPolicy.retry(requestOptions, err, response, body)) {
+            const backoff = backoffs[retries];
+            retries += 1;
+            setTimeout(requestOnce, backoff);
+            return;
+          }
+          if (err) {
+            rejecter(err);
+            cbW(err);
+            return;
+          }
+
+          let e;
+          switch (response.statusCode) {
+            case 200:
+              if (saveResults) {
+                results = results.concat(body.authorSet.results.map(f));
+              } else {
+                body.authorSet.results.forEach(f);
+              }
               break;
+            
+            case 400:
+              e = new Errors.BadRequest(body || {});
+              rejecter(e);
+              cbW(e);
+              return;
+            
+            case 500:
+              e = new Errors.InternalError(body || {});
+              rejecter(e);
+              cbW(e);
+              return;
+            
             default:
-              rejecter(new Error("Recieved unexpected statusCode " + response.statusCode));
+              e = new Error("Recieved unexpected statusCode " + response.statusCode);
+              rejecter(e);
```

Comment:
Would it make more sense to call this rejector in the cbWhile callback? Seems like it would make it cleaner to read.